testing/sdk_upgrade.py

```diff
@@ -147,49 +147,61 @@ def get_config(package_name, service_name):
     return target_config
 
 
-def update_service(package_name, service_name, additional_options=None, to_package_version=None):
-    update_cmd = ['update', 'start']
-    if to_package_version:
-        update_cmd.append('--package-version={}'.format(to_package_version))
-    if additional_options:
-        options_file = tempfile.NamedTemporaryFile("w")
-        json.dump(additional_options, options_file)
-        options_file.flush()  # ensure json content is available for the CLI to read below
-        update_cmd.append("--options={}".format(options_file.name))
-    sdk_cmd.svc_cli(package_name, service_name, ' '.join(update_cmd), check=True)
-
-
-def _upgrade_or_downgrade(
-        package_name,
-        to_package_version,
-        service_name,
-        running_task_count,
-        additional_options,
-        timeout_seconds,
-        wait_for_deployment):
-
-    initial_config = get_config(package_name, service_name)
-    task_ids = sdk_tasks.get_task_ids(service_name, '')
-
+def update_service(package_name,
```

Comment:
We have made this interface more complex by adding options that are not used at all.

I would:
1. Remove the 1.9 cluster specifics from here.
2. Remove the additional options from the `update_service` function.
3. Still call `update_service` from `_upgrade_or_downgrade`
4. Add a check with a visible error message when `update_service` is called on a 1.9 cluster.

We could drop the check for `1.9` clusters if we set `expected_running_tasks=0` and `wait_for_deployment=False` when calling `sdk_install.install` and then perform the equivalent checks in `_upgrade_or_downgrade`.

Reply:
The only place this method is used needs it work for 1.9 as well. If i remove that then this method would be unused. I can remove this method and increase the visibility of `upgrade_or_downgrade`.

Reply:
For clarification, the issue was not that this was not running correctly against 1.9, it was that this was not running correctly against **1.10 Open**.